### ISSUE TITLE
We need to support multiple chains including testnets

### ISSUE THREAD

**Issue by tjayrush** [TrueBlocks/trueblocks-core#1733]
(1) we need to provide a place in our cache and unchained folders to handle other chains. This is easy as all access to both of those folders are in a single function and we can easily add the chainid to those paths. (We can query the chain ids -- similar to what's described here: https://besu.hyperledger.org/en/stable/Concepts/NetworkID-And-ChainID/). In this way, the user won't have to do anything.

(2) In addition to where we store data (in which cache), there's the issue of where we get data (as we learned with the timestamps file and the manifest file for chifra init). These also can be store per chain using the chain id.

**Comment by talespa** [comment]
I'm testing on Ropsten (id: 3) but when i try to run '$ chifra init' i have a FATAL like this:

```
~/trueblocks-core$ chifra init
INFO[18-11|15:11:17.902] hashToIndexFormatFile: Qmart6XP9XjL43p72PGR93QKytbK8jWWcMguhFgxATTya2
INFO[18-11|15:11:17.902] hashToBloomFormatFile: QmNhPk39DUFoEdhUmtGARqiFECUHeghyeryxZM9kyRxzHD
INFO[18-11|15:11:17.902] unchainedIndexAddr: 0xcfd7f3b24f3551741f922fd8c4381aa4e00fc8fd
INFO[18-11|15:11:17.902] manifestHashEncoding: 0x337f3f32
INFO[18-11|15:11:17.902] Calling unchained index smart contract...
FATAL[18-11|15:11:18.617] while unpacking value: abi: attempting to unmarshall an empty string while arguments are expected
```

**Comment by tjayrush** [comment]
So sorry for your trouble. This is definitely a bug in our code. 

One of the problems we have is that we don't normally test against Ropsten but only because of a lack of resources. I'll mark this as a priority, and try to fix it for you over the next weekend.

**Comment by zimbabao** [comment]
```
ubuntu@:/web3-index/code/trueblocks-core (master)$ ./bin/chifra init
INFO[23-11|23:19:40.310] hashToIndexFormatFile: Qmart6XP9XjL43p72PGR93QKytbK8jWWcMguhFgxATTya2
INFO[23-11|23:19:40.310] hashToBloomFormatFile: QmNhPk39DUFoEdhUmtGARqiFECUHeghyeryxZM9kyRxzHD
INFO[23-11|23:19:40.310] unchainedIndexAddr: 0xcfd7f3b24f3551741f922fd8c4381aa4e00fc8fd
INFO[23-11|23:19:40.311] manifestHashEncoding: 0x337f3f32
INFO[23-11|23:19:40.311] Calling unchained index smart contract...
FATAL[23-11|23:19:40.318] while unpacking value: abi: attempting to unmarshall an empty string while arguments are expected
```

I see this issue on ubuntu  18.04.1

**Comment by somecryptodude** [comment]
I see this same issue on BSC, and I updated all references from Etherscan's API to bscscan's API.  Thanks

**Comment by TheGreatAxios** [comment]
Is there any update on this? When using locally with a custom side chain this error is duplicated.

Thanks!

**Comment by tjayrush** [comment]
We just released v0.22.0-alpha which was a lot of work related to the performance of the names database. I'm debating what's next...

The major problem for us is one of resources. We're severely limited. Without explicit financial support, I have a feeling this issue will lag quite substantially other than support for the merge and Eth 2.0 which we will be forced to support.

It would be a massive help to us if those interested could help us categorize the issues. We know of a few already (names, indexes, determining tracing). But, the biggest issue, is setting up and gaining access to the RPC of these other chains. We run our own Ethereum mainnet clients, and that, in and of itself, is quite a lot of work.

We would hire-out other endpoints, but the way TrueBlocks hits the node, we get almost instantaneously rate-limited and we can't afford to hire-out dedicated servers.

We're open to solutions...

**Comment by tjayrush** [comment]
We have these issues all combined in multichain.md file. Okay to close.
